### ISSUE TITLE
fold learned theory

### ISSUE THREAD

**Issue by andrewcropper** [metagol/metagol#2]
If we learn a theory in which an invented predicate is used only once, then it would be nice to have a 'fold_result' option.

For instance, given the theory T:

grandmother(A,B):- female(A), grandmother_1(A,B).
grandmother_1(A,B):- parent(A,C), parent(C,B).

Metagol folds the program and prints the theory T':

grandmother(A,B):- female(A), parent(A,C), parent(C,B).

However, it should not fold the program when an invented predicate is used more than once. For instance:

gg_grandparent(A,B):- gg_grandparent_1(A,C), gg_grandparent_1(C,B).
gg_grandparent_1(A,B):- parent(A,C), parent(C,B).
